File: mobile-app/lib/features/services/domain/service_model.dart
class HotelService {
  final String id;
  final String name;
  final String description;
  final double price;
  final String category;
  final String image;
  final bool isAvailable;
  final int? duration;

  HotelService({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.category,
    required this.image,
    required this.isAvailable,
    this.duration,
  });

  factory HotelService.fromJson(Map<String, dynamic> json) {
    return HotelService(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      category: json['category'] ?? 'other',
      image: json['image'] ?? '',
      isAvailable: json['isAvailable'] ?? true,
      duration: json['duration'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'description': description,
      'price': price,
      'category': category,
      'image': image,
      'isAvailable': isAvailable,
      'duration': duration,
    };
  }
}

File: backend/src/models/service.model.js
const mongoose = require('mongoose');

const serviceSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: [true, 'Please add a service name'],
      trim: true,
      unique: true,
    },
    description: {
      type: String,
      required: [true, 'Please add a description'],
      maxlength: [500, 'Description can not be more than 500 characters'],
    },
    price: {
      type: Number,
      default: 0,
      min: [0, 'Price must be greater than or equal to 0'],
    },
    pricingNote: {
      type: String,
      trim: true,
      maxlength: [100, 'Pricing note can not be more than 100 characters'],
    },
    category: {
      type: String,
      enum: ['spa', 'gym', 'pool', 'laundry', 'airport_transfer', 'concierge', 'room_service', 'other'],
      default: 'other',
    },
    section: {
      type: String,
      enum: ['hotel_service', 'wellness', 'general'],
      default: 'hotel_service',
    },
    badge: {
      type: String,
      trim: true,
      maxlength: [50, 'Badge can not be more than 50 characters'],
    },
    icon: {
      type: String,
      trim: true,
    },
    image: {
      type: String,
      default: 'default-service.jpg',
    },
    imagePublicId: {
      type: String,
    },
    isAvailable: {
      type: Boolean,
      default: true,
    },
    duration: {
      type: Number,
    },
  },
  {
    timestamps: true,
  }
);

serviceSchema.index({ section: 1, isAvailable: 1 });

module.exports = mongoose.model('Service', serviceSchema);
